### PR TITLE
YARD extension for slots

### PR DESCRIPTION
**Pull request by manuelpuyol**
Closes https://github.com/primer/view_components/issues/188

- Adds a way to document slots by creating handlers for `renders_one` and `renders_many` calls.
- Adds a method to create links between components in the doc.
- Adds documentation to `ButtonGroup#buttons` slot

image

### CODE REVIEW

File: app/components/primer/button_group_component.rb [hunk 1]
```diff
@@ -5,6 +5,9 @@ module Primer
   class ButtonGroupComponent < Primer::Component
     include ViewComponent::SlotableV2
 
+    # Required list of buttons to be rendered.
+    #
+    # @param kwargs [Hash] the same arguments as <%= link_to_component(Primer::ButtonComponent) %>
```
Comment (joelhawksley) [inline app/components/primer/button_group_component.rb:10]: What do you think having this import all of the arguments from `ButtonComponent`, so we don't have to link to it?

Comment (manuelpuyol) [reply app/components/primer/button_group_component.rb:10]: we can do it as a next step! I don't want this PR to become overcomplicated

Comment (joelhawksley) [inline app/components/primer/button_group_component.rb:10]:
```suggestion
    # @param kwargs [Hash] The same arguments as <%= link_to_component(Primer::ButtonComponent) %>.
```

**Review by joelhawksley — APPROVED** [review]
Cool use of YARD here!

**Pull request merged**